fili-core/src/main/java/com/yahoo/bard/webservice/data/config/dimension/DefaultKeyValueStoreDimensionConfig.java

```diff
@@ -0,0 +1,145 @@
+    private final String category;
+    private final LinkedHashSet<DimensionField> fields;
+    private final LinkedHashSet<DimensionField> defaultDimensionFields;
+    private final KeyValueStore keyValueStore;
+    private final SearchProvider searchProvider;
+
+    /**
+     * Construct a DefaultKeyValueStoreDimensionConfig instance from dimension name, dimension fields and
+     * default dimension fields.
+     *
+     * @param apiName  The API Name is the external, end-user-facing name for the dimension.
+     * @param physicalName  The internal, physical name for the dimension.
+     * @param description  A description of the dimension and its meaning.
+     * @param longName  The Long Name is the external, end-user-facing long  name for the dimension.
+     * @param category  The Category is the external, end-user-facing category for the dimension.
+     * @param fields  The set of fields for this dimension.
+     * @param defaultDimensionFields  The default set of fields for this dimension to be shown in the response.
+     * @param keyValueStore  The key value store holding dimension row data.
+     * @param searchProvider  The search provider for field value lookups on this dimension.
+     */
+    public DefaultKeyValueStoreDimensionConfig(
+        @NotNull DimensionName apiName,
+        String physicalName,
+        String description,
+        String longName,
+        String category,
+        @NotNull LinkedHashSet<DimensionField> fields,
+        @NotNull LinkedHashSet<DimensionField> defaultDimensionFields,
+        @NotNull KeyValueStore keyValueStore,
+        @NotNull SearchProvider searchProvider
+    ) {
+      this.apiName = apiName;
+      this.physicalName = physicalName;
+      this.description = description;
+      this.longName = longName;
+      this.category = category;
+      this.fields = fields;
+      this.defaultDimensionFields = defaultDimensionFields;
+      this.keyValueStore = keyValueStore;
+      this.searchProvider = searchProvider;
+    }
+
+  /**
+   * Construct a DefaultKeyValueStoreDimensionConfig instance from dimension name and
+   * only using default dimension fields.
+   *
+   * @param apiName  The API Name is the external, end-user-facing name for the dimension.
+   * @param physicalName  The internal, physical name for the dimension.
+   * @param description  A description of the dimension and its meaning.
+   * @param longName  The Long Name is the external, end-user-facing long  name for the dimension.
+   * @param category  The Category is the external, end-user-facing category for the dimension.
+   * @param defaultDimensionFields  The default set of fields for this dimension to be shown in the response.
+   * @param keyValueStore  The key value store holding dimension row data.
+   * @param searchProvider  The search provider for field value lookups on this dimension.
+   */
+  public DefaultKeyValueStoreDimensionConfig(
+      DimensionName apiName,
```

Comment:
Does it make sense that the other constructor requires some of these arguments to be `@NotNull` but this constructor does not require? And we should just put `@NotNull` for all the arguments since thats easier. Apply to other files too.